homeassistant/helpers/dispatcher.py

```diff
@@ -62,7 +62,9 @@ def dispatcher_send(hass: HomeAssistant, signal: str, *args: Any) -> None:
     hass.loop.call_soon_threadsafe(async_dispatcher_send, hass, signal, *args)
 
 
-def _generate_job(signal: str, target: Callable[..., Any]) -> HassJob:
+def _generate_job(
+    signal: str, target: Callable[..., Any]
+) -> HassJob[..., Coroutine[Any, Any, Any]]:
```

Comment:
The target is not always a coroutine function.

Reply:
Thanks! Yeah it should indeed be `HassJob[..., *None* | Coroutine[Any, Any, *None*]` as returned by `catch_log_exception`.